cloud/docker/docker_login.py

```diff
@@ -0,0 +1,243 @@
+
+######################################################################
+
+DOCUMENTATION = '''
+---
+module: docker_login
+author: Olaf Kilian
+version_added: "1.9"
+short_description: Manage Docker registry logins
+description:
+     - Ansible version of the "docker login" CLI command.
+     - This module allows you to login to a Docker registry without directly pulling an image or performing any other actions.
+     - It will write your login credentials to your local .dockercfg file that is compatible to the Docker CLI client as well as docker-py and all other Docker related modules that are based on docker-py.
+options:
+  registry:
+    description:
+       - URL of the registry, for example: https://index.docker.io/v1/
+    required: true
+    default: null
+    aliases: []
+  username:
+    description:
+       - The username for the registry account
+    required: true
+    default: null
+    aliases: []
+  password:
+    description:
+       - The plaintext password for the registry account
+    required: true
+    default: null
+    aliases: []
+  email:
+    description:
+       - The email address for the registry account
+    required: false
+    default: None
+    aliases: []
+  reauth:
+    description:
+       - Whether refresh existing authentication on the Docker server (boolean)
+    required: false
+    default: false
+    aliases: []
+  dockercfg_path:
+    description:
+       - Use a custom path for the .dockercfg file
+    required: false
+    default: ~/.dockercfg
+    aliases: []
+  docker_url:
+    descriptions:
+       - Refers to the protocol+hostname+port where the Docker server is hosted
+    required: false
+    default: unix://var/run/docker.sock
+    aliases: []
+  timeout:
+    description:
+       - The HTTP request timeout in seconds
+    required: false
+    default: 600
+    aliases: []
+
+requirements: [ "docker-py" ]
+'''
+
+EXAMPLES = '''
+Login to a Docker registry without performing any other action. Make sure that the user you are using is either in the docker group which owns the Docker socket or use sudo to perform login actions:
+
+- name: login to DockerHub remote registry using your account
+  docker_login:
+    username: docker
+    password: rekcod
+    email: docker@docker.io
+
+- name: login to private Docker remote registry and force reauthentification
+  docker_login:
+    registry: https://your.private.registry.io/v1/
+    username: yourself
+    password: secrets3
+    reauth: yes
+
+- name: login to DockerHub remote registry using a custom dockercfg file location
+  docker_login:
+    username: docker
+    password: rekcod
+    email: docker@docker.io
+    dockercfg_path: /tmp/.mydockercfg
+
+'''
+
+try:
+    import os.path
+    import sys
+    import json
+    import base64
+    import docker.client
+    from requests.exceptions import *
+    from urlparse import urlparse
+except ImportError, e:
+    print "failed=True msg='failed to import python module: %s'" % e
+    sys.exit(1)
```

Comment:
Do not use print and sys.exit in modules. Look at this example here  https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/cloudstack/cs_affinitygroup.py#L95
and here https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/cloudstack/cs_affinitygroup.py#L209
how you fail with module.fail_json if a python module is missing